pkg/framework/provider/crypto/crypto.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+// Package crypto provides the default implementation of pkg/framework/spi/crypto.Crypto interface
+//
+// It uses github.com/tink/go crypto primitives
+package crypto
+
+import (
+	"errors"
+
+	"github.com/google/tink/go/aead"
+	"github.com/google/tink/go/core/primitiveset"
+	"github.com/google/tink/go/keyset"
+	"github.com/google/tink/go/signature"
+	aeadsubtle "github.com/google/tink/go/subtle/aead"
+	"golang.org/x/crypto/chacha20poly1305"
+)
+
+// Package provider/crypto includes implementation of spi/crypto. SPI implementation will be built
+// as a Framework option and fed into pkg/common/crypto implementation that includes a combined crypto
+// and kms API to be used everywhere by the framework.
+
+// TinkCrypto is the default Crypto SPI implementation using Tink
+type TinkCrypto struct {
+}
+
+// New creates a new TinkCrypto instance
+func New() (*TinkCrypto, error) {
+	return &TinkCrypto{}, nil
+}
+
+// Encrypt will encrypt msg using the implementation's corresponding encryption key and primitive in kh
+func (t *TinkCrypto) Encrypt(msg, aad []byte, kh interface{}) ([]byte, []byte, error) {
+	keyHandle, ok := kh.(*keyset.Handle)
+	if !ok {
+		return nil, nil, errors.New("bad key handle format")
+	}
+
+	a, err := aead.New(keyHandle)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	ct, err := a.Encrypt(msg, aad)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	ps, err := keyHandle.Primitives()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Tink appends a key prefix + nonce to ciphertext, let's remove them to get the raw ciphertext
+	ivSize := nonceSize(ps)
+	prefixLength := len(ps.Primary.Prefix)
+	cipherText := ct[prefixLength+ivSize:]
+	nonce := ct[prefixLength : prefixLength+ivSize]
+
+	return cipherText, nonce, nil
+}
+
+func nonceSize(ps *primitiveset.PrimitiveSet) int {
+	var ivSize int
+	// AESGCM and XChacha20Poly1305 nonce sizes supported only for now
+	switch ps.Primary.Primitive.(type) {
+	case *aeadsubtle.XChaCha20Poly1305:
+		ivSize = chacha20poly1305.NonceSizeX
+	default:
```

Comment:
I still prefer having `case aeadsubtle.AESGCM...` explicitly.

Reply:
sure, I'll add it ..